### PR TITLE
Add iml-orm crate

### PR DESCRIPTION
**Pull request by jgrund**
Add a new crate `iml-orm` that can be used with
[diesel](http://diesel.rs/) to make working with the db much simpler.

The schema has been generated from the IML Db, and structs have been
generated to represent each table in the schema.

The model structs may be used in other contexts for example both the
iml-agent and iml-gui.

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

### CODE REVIEW

**Comment by mkpankov** [comment]
What's the async story with Diesel?

I believe some time ago they wasn't async and weren't planning. Are we going to work with it via a thread pool?

**Comment by jgrund** [comment]
> What's the async story with Diesel?

They aren't async and there are some fundamental issues getting there.

> Are we going to work with it via a thread pool?

[yes](https://github.com/mehcode/tokio-diesel)

File: iml-manager-env/src/lib.rs [hunk 1]
```diff
@@ -138,3 +138,27 @@ pub fn get_db_name() -> Option<String> {
...
+
+    if let Some(x) = get_db_name() {
+        xs.push(format!("dbname={}", x));
+    }
+
+    if let Some(x) = get_db_password() {
+        xs.push(format!("password={}", x));
+    }
+
+    let s = xs.join(" ");
```
Comment (johnsonw) [inline iml-manager-env/src/lib.rs]: Could do:
```
xs.join(" ")
```

Comment (jgrund) [reply iml-manager-env/src/lib.rs]: yeah, missed that lint.

File: iml-orm/src/schema.rs [hunk 2]
```diff
@@ -0,0 +1,1907 @@
+table! {
```
Comment (mkpankov) [inline iml-orm/src/schema.rs]: Is this file written by hand?

Comment (jgrund) [reply iml-orm/src/schema.rs]: This is generated from a production db.

File: iml-manager-env/src/lib.rs [hunk 3]
```diff
@@ -138,3 +138,25 @@ pub fn get_db_name() -> Option<String> {
...
     empty_str_to_none(get_var("DB_PASSWORD"))
 }
+
+/// Gets a connection string from the IML env
+pub fn get_db_conn_string() -> String {
+    let mut xs = vec![format!("user={}", get_db_user())];
+
+    let host = match get_db_host() {
+        Some(x) => x,
+        None => "/var/run/postgresql".into(),
```
Comment (nlinker) [inline iml-manager-env/src/lib.rs]: Never seen host like this before...

**Pull request merged**